src/components/verify/verify-page.tsx

```diff
@@ -57,10 +60,41 @@ export const VerifyPage: React.FunctionComponent = () => {
   const [rawDocument, setRawDocument] = useState<WrappedDocument<v2.OpenAttestationDocument>>();
   const [issuer, setIssuer] = useState("");
   // set overall status to idle and set the rest to pending
-  const [verificationStatus, setVerificationStatus] = useState<Status>(Status.IDLE);
+  const [loadDocumentStatus, setLoadDocumentStatus] = useState(Status.IDLE);
+  const [loadDocumentError, setLoadDocumentError] = useState("");
+  const [verificationStatus, setVerificationStatus] = useState(Status.IDLE);
   const [issuerStatus, setIssuerStatus] = useState(Status.PENDING);
   const [issuingStatus, setIssuingStatus] = useState(Status.PENDING);
   const [tamperedStatus, setTamperedStatus] = useState(Status.PENDING);
+  const location = useLocation();
+  // use layout effect to run this as soon as possible otherwise the dropzone might be displayed before disappearing
+  useLayoutEffect(() => {
+    const run = async () => {
+      try {
+        if (location.search) {
+          const parsedSearch = queryString.parse(location.search);
+          if (typeof parsedSearch.q !== "string") {
+            return;
+          }
+          const action = JSON.parse(window.decodeURI(parsedSearch.q));
+
+          setLoadDocumentStatus(Status.PENDING);
+          const WAIT = 2000;
+          const [wrappedDocument] = await Promise.all([retrieveDocument(action), wait(WAIT)]);
+          setLoadDocumentStatus(Status.REJECTED);
```

Comment:
is this supposed to be `Status.RESOLVED`?

Reply:
yup, changed :)